sui_core/src/authority.rs

```diff
@@ -103,6 +195,8 @@ pub struct AuthorityState {
 
     /// Ensures there can only be a single consensus client is updating the state.
     pub consensus_guardrail: AtomicUsize,
+
+    pub metrics: Arc<AuthorityMetrics>,
```

Comment:
ok i sort of see what you're going for here.

Reply:
I suppose if we do it this way we could actually avoid registering metrics with a global aggregator and instead register metrics with just the aggregator for the authority. (ie avoid relying on global state to begin with and allow for running multiple authorities in the same process without them clobbering each others metrics)

Reply:
Bonus points if we can construct metrics with a custom [Registry](https://docs.rs/prometheus/latest/prometheus/struct.Registry.html) and avoid using a global one

Reply:
We could use a custom registry for each one, however I'm not sure I see the point of it, at least in terms of exposing a separate metrics endpoint for each authority...   that is typically a production concern where we would not have multiple authorities in a single process.   I actually wish the "multiple authorities in a single process" thing would go away, as it distorts how they run in a real cluster or environment.   I suppose we can separate into different registries and then worry about how to expose it if people want separately.